pkg/operator/starter.go

```diff
@@ -142,8 +142,21 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 		controllerContext.EventRecorder,
 	)
 
+	versionRecorder := status.NewVersionGetter()
+	clusterOperator, err := configClient.ConfigV1().ClusterOperators().Get(ctx, "authentication", metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	// perform version changes to the version getter prior to tying it up in the status controller
+	// via change-notification channel so that it only updates operator version in status once
+	// either of the workloads synces
+	for _, version := range clusterOperator.Status.Versions {
+		versionRecorder.SetVersion(version.Name, version.Version)
+	}
+	versionRecorder.SetVersion("operator", os.Getenv("OPERATOR_IMAGE_VERSION"))
```

Comment:
should this be set by `NewOAuthServerWorkloadController` ?

Reply:
no, that's the operator's version

Reply:
to be precise: the change in operator version will only actually happen once either of the workload controllers synces their version.